presto-main/src/main/java/io/prestosql/execution/CreateMaterializedViewTask.java

```diff
@@ -0,0 +1,147 @@
+import io.prestosql.Session;
+import io.prestosql.connector.CatalogName;
+import io.prestosql.execution.warnings.WarningCollector;
+import io.prestosql.metadata.Metadata;
+import io.prestosql.metadata.QualifiedObjectName;
+import io.prestosql.security.AccessControl;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.connector.ConnectorMaterializedViewDefinition;
+import io.prestosql.sql.analyzer.Analysis;
+import io.prestosql.sql.analyzer.Analyzer;
+import io.prestosql.sql.analyzer.FeaturesConfig;
+import io.prestosql.sql.parser.SqlParser;
+import io.prestosql.sql.tree.CreateMaterializedView;
+import io.prestosql.sql.tree.Expression;
+import io.prestosql.sql.tree.NodeRef;
+import io.prestosql.sql.tree.Parameter;
+import io.prestosql.sql.tree.Statement;
+import io.prestosql.transaction.TransactionManager;
+
+import javax.inject.Inject;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.util.concurrent.Futures.immediateFuture;
+import static io.prestosql.metadata.MetadataUtil.createQualifiedObjectName;
+import static io.prestosql.spi.StandardErrorCode.NOT_FOUND;
+import static io.prestosql.spi.connector.ConnectorViewDefinition.ViewColumn;
+import static io.prestosql.sql.NodeUtils.mapFromProperties;
+import static io.prestosql.sql.ParameterUtils.parameterExtractor;
+import static io.prestosql.sql.SqlFormatterUtil.getFormattedSql;
+import static java.util.Objects.requireNonNull;
+
+public class CreateMaterializedViewTask
+        implements DataDefinitionTask<CreateMaterializedView>
+{
+    private final SqlParser sqlParser;
+
+    @Inject
+    public CreateMaterializedViewTask(SqlParser sqlParser, FeaturesConfig featuresConfig)
+    {
+        this.sqlParser = requireNonNull(sqlParser, "sqlParser is null");
+        requireNonNull(featuresConfig, "featuresConfig is null");
+    }
+
+    @Override
+    public String getName()
+    {
+        return "CREATE MATERIALIZED VIEW";
+    }
+
+    @Override
+    public String explain(CreateMaterializedView statement, List<Expression> parameters)
+    {
+        return "CREATE MATERIALIZED VIEW " + statement.getName();
+    }
+
+    @Override
+    public ListenableFuture<?> execute(CreateMaterializedView statement, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, QueryStateMachine stateMachine, List<Expression> parameters)
+    {
+        Session session = stateMachine.getSession();
+        QualifiedObjectName name = createQualifiedObjectName(session, statement, statement.getName());
+        Map<NodeRef<Parameter>, Expression> parameterLookup = parameterExtractor(statement, parameters);
+
+        accessControl.checkCanCreateView(session.toSecurityContext(), name);
+        accessControl.checkCanCreateTable(session.toSecurityContext(), name);
```

Comment:
Why both?

Reply:
Also, permissions are being checked in the analyzer, as well, so this is redundant

Reply:
You are right, the permissions are being checked in Analyzer. So, I will remove this code. 
I think here is how access checks should be done (current code is not complete/correct .. will fix). 
+ StatementAnalyzer:visitCreateMaterializedView:
   + Check 'canCreateView' with materialized view name as parameter
   + Check 'canCreateTable' with storage table name as parameter
+ DropMaterializedViewTask:execute:
   + Check 'canDropView' with materialized view name as parameter
   + check 'canDropTable' with storage table name as parameter
+ Refresh materialized view is broken into delete/insert etc .. so those commands perform access control checks.
What do you think?

Reply:
We talked offline about credential passthrough. It does not work as expected for views as well and materialized views emulate the view design. So, for now, we will document that credential passthrough is not supported with materialized views.